src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocNodeImpl.java

```diff
@@ -172,16 +230,46 @@ public void setIndex(int index) {
         this.index = index;
     }
 
+    /**
+     * Sets next sibling node.
+     *
+     * @param nextSibling Next sibling node.
+     */
+    public void setNextSibling(DetailNode nextSibling) {
+        this.nextSibling = (JavadocNodeImpl) nextSibling;
```

Comment:
Not sure why we accept `DetailNode` as a parameter and convert it to `JavadocNodeImpl`, but `setParent` accepts `JavadocNodeImpl` and we must convert outside callers from `DetailNode`.
There seems to be no consistency.